GlobalConstantsService/src/ParticleDataList.cc

```diff
@@ -0,0 +1,121 @@
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+#include "cetlib_except/exception.h"
+
+#include "Offline/GlobalConstantsService/inc/ParticleDataList.hh"
+#include "Offline/ConfigTools/inc/ConfigFileLookupPolicy.hh"
+#include "Offline/ConfigTools/inc/SimpleConfig.hh"
+#include "Offline/DataProducts/inc/PDGCode.hh"
+
+
+namespace mu2e {
+
+  // *************************************************************
+
+  ParticleDataList::ParticleDataList( SimpleConfig const& config ) {
+
+    ConfigFileLookupPolicy findConfig;
+    
+    std::string tableFilename = findConfig(
+                    config.getString("particleDataList.filename",
+                   "Offline/GlobalConstantsService/data/ParticleList.txt") );
+    if( tableFilename.empty() ) {
+      throw cet::exception("PARTICLELIST_NO_FILE") 
+        << "ParticleDataList: file name not found\n";
+    }
+
+    std::ifstream in(tableFilename.c_str());
+    if ( !in ) {
+      throw cet::exception("PARTICLELIST_OPEN_FAIL")
+          << "Unable to open particle data file: "
+          << tableFilename << "\n";
+    }
+
+    std::string line;
+    std::string str;
+    std::vector<std::string> words;
```

Comment:
I suggest

constexpr int exepectedWords = 7;
words.reserve(expectedWords);

and replace the literal 7 everywhere with this constexpr

Reply:
Thanks.